### ISSUE TITLE
Bottom toolbar/address bar crashes on Android v1.73.89

### ISSUE THREAD

**Issue by igorskyflyer** [brave/brave-browser#42347]
### Description

In the previous version I changed a setting in the `brave://flags` page called `Bottom toolbar` in attempt to have a bottom toolbar/address bar, of course. That setting enabled a menu entry in the Settings that allows the user to set the toolbar position, see the screenshots below.

At the time, the toolbar was still at the top, even when set to Bottom. Unfortunately, I did not revert the given setting and the browser worked fine but after updating to the latest version, Brave now actually has a bottom toolbar but it also crashes indefinitely. How can I reset the flags without losing all my browsing data?

I have tried launching many activities, all of them crash except WebDev but nothing can be done from there. I've tried using a Search widget to trick Brave into launching a predefined URL (brave://flags) no luck, still crashes immediately.

### Steps to reproduce

1. Be on a previous version on Brave (`v1.71.123`) on Android
2. Open `brave://flags`
3. Set `Bottom Toolbar` to **enabled** (1)
4. Restart
5. Open the app's Settings
6. Click on `Address bar`
7. Set to **Bottom** (2)
8. Install the latest version (`v1.73.89`)  

Relevant screenshots

<div align="center">
  <figure>
    <img src="https://github.com/user-attachments/assets/f85a0bc4-5804-41a1-acaf-36d2f1b397a4" alt="Screenshot">
    <figcaption><em>Figure 1. Brave's flags page</em></figcaption>
  </figure>
  <br>
  <br>
    <figure>
    <img src="https://github.com/user-attachments/assets/7f35f653-08e5-4fd4-a588-cf5541fba2fb" alt="Screenshot">
    <figcaption><em>Figure 2. Address bar position</em></figcaption>
</div>

### Actual result

The browser crashes indefinitely.

### Expected result

No crashes should occur.

### Reproduces how often

Easily reproduced

### Brave version

v1.73.89

### Device

- Brand/model: Samsung A55
- Android version: 14

### Channel information

- [x] release (stable)
- [ ] beta
- [ ] nightly

### Reproducibility

- [x] with Brave Shields disabled
- [x] with Brave Rewards disabled
- [ ] in the latest version of Chrome

**Comment by SlowNicoFish** [comment]
This needs to be integrated better. It's the only big thing that brave misses in my opinion

**Comment by igorskyflyer** [comment]
> This needs to be integrated better. It's the only big thing that brave is missing in my opinion

Indeed, if it wasn't for this lock-out that happened to me I would still be tinkering how to make the toolbar appear at the bottom, all browsers have it, even Chrome.

**Comment by igorskyflyer** [comment]
Pinpointed the actual issue, caused by the Ad-block engine actually. The bug was introduced in the v1.73.xxx version.
